### PR TITLE
Add deep symbolization to CustomButton `visibility` field

### PR DESCRIPTION
**Pull request by AparnaKarve**
The backend expects the `CustomButton` `visibility` field to be deep symbolized in order to perform Custom Button related operations.

Hence appropriate adjustment had to be made in the API as well.

### CODE REVIEW

**Comment by AparnaKarve** [comment]
This fix is required for https://github.com/ManageIQ/manageiq-ui-classic/pull/2569

**Comment by abellotti** [comment]
Thanks @AparnaKarve this is fine, is this needed for gaprindashvili or just master ?

**Comment by AparnaKarve** [comment]
@abellotti This is needed for gaprindashvili

File: spec/requests/custom_buttons_spec.rb [hunk 1]
```diff
@@ -85,18 +87,20 @@
...
       expected = {
         'results' => a_collection_including(
-          a_hash_including('id' => cb.id.to_s, 'name' => 'updated 1'),
+          a_hash_including('id' => cb.id.to_s, 'name' => 'updated 1', 'visibility' => {'roles' => ['_ALL_']}),
         )
       }
+      custom_button = CustomButton.find(response.parsed_body['results'].first["id"])
       expect(response).to have_http_status(:ok)
       expect(response.parsed_body).to include(expected)
+      expect(custom_button.visibility[:roles]).to eq(['_ALL_'])
```
Comment (abellotti) [inline spec/requests/custom_buttons_spec.rb]: would prefer that we check the custom_button we have to start with, maybe just expect(cb.reload.visibility[:roles]).to ....

Comment (AparnaKarve) [reply spec/requests/custom_buttons_spec.rb]: I have addressed this in the last commit (5cb4355)

Comment (AparnaKarve) [reply spec/requests/custom_buttons_spec.rb]: commit revised further - 6d268dc

**Comment by AparnaKarve** [comment]
@abellotti Thank you!

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit d03bc5da2280a46685053013071d0855a0b402f5
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Mon Nov 13 16:44:25 2017 -0500

    Merge pull request #206 from AparnaKarve/add_deep_symbolizations_to_cb_visibility
    
    Add deep symbolization to CustomButton `visibility` field
    (cherry picked from commit 85c938362984af081d6eb16e92fc874ae471e09c)
```

**Pull request merged**